File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
 
 int a = 0;
 int k = 0;
 int b = 0;
 
    if (argc != 2)
    {
     printf("usage: ./caesar k\n");
     return 1;
    }
    
    for(int i = 0, n = strlen(argv[1]); i < n; i++)
    {
     
     a = (int) argv[1][i] - '0';
     
     k = k * 10 + a;
    }

    string plaintext;
    
    printf("plaintext: ");
    plaintext = get_string();
    
    printf("ciphertext: ");
    
   for(int i = 0, n = strlen(plaintext); i < n; i++)
   {
        b = (int) plaintext[i];
        if(b >= 65 && b <= 90)
        {
            int p = (int) plaintext[i] - 65;
            int c =(p + k) % 26;
            int j = c + 65;
            printf("%c", (char) j);
        }
        else if(b >= 97 && b <= 122)
        {
            int p = (int) plaintext[i] - 97;
            int c = (p + k) % 26;
            int j = c + 97;
            printf("%c", (char) j);
        }
        else
        {
         printf("%c", b);
        }
   }
   printf("\n");
}  

File: pset6/caesar.py
import sys

def cc(k, plaintext):
    print("ciphertext: ", end="")
    for i in range(0, textlength):
        
        if plaintext[i].isupper():
            int_char = ((((ord(plaintext[i])) - 65) + k) % 26) + 65
            char = chr(int_char)
            print("%c"% (char), end="")
            
        elif plaintext[i].islower():
            int_char = ((((ord(plaintext[i]) - 97) + k) % 26) + 97)
            char = chr(int_char)
            print("%c"% (char), end="")
            
        else:
            print("%c"% (plaintext[i]), end="")

    print("")
    
if len(sys.argv) > 1:
    k = int(str(sys.argv[1]))
    if k >= 0:
        plaintext = input("plaintext: ")
        textlength = len(plaintext)
        cc(k, plaintext)
    else:
        print("Try again using an integer as key")
else:
    print("usage: python caesar.py k")    